### ISSUE TITLE
Tag with shortclosing and with greater than in attribute parsing problem

### ISSUE THREAD

**Issue by stefanpausch** [prewk/xml-string-streamer#60]
Problem: If an greater than (>) sign (non html entity formatted) exists within at short closing tag attribute the Streamer cannot find the next matching tag and reads until the the end of file, which causes trouble in parsing the returned node with simplexml.

Running xml-string-streamer 0.11.0

Example code:

`<?xml version="1.0" encoding="UTF-8"?>
<root>
  <child addressLine3="teststring>"/>
  <child addressLine3="teststring"/>
  <child addressLine3="teststring"/>
</root>
`
With the example code the streamer finds the first child and returns ALL xml afterwards (does not find the 2nd/3rd child). 

If i configure the StringWalker with expectGT and Short Closing Tag the Parser does not find any nodes at all:

`expectGT' => true,
'tagsWithAllowedGT' => array(
array("<", "/>"),
),`

As i read greater than is a non valid character, but within attributes it is allowed. DOMReader and SimpleXML parse the file without any  problems.

`SimpleXMLElement Object
(
    [child] => Array
        (
            [0] => SimpleXMLElement Object
                (
                    [@attributes] => Array
                        (
                            [addressLine3] => teststring>
                        )

                )

            [1] => SimpleXMLElement Object
                (
                    [@attributes] => Array
                        (
                            [addressLine3] => teststring
                        )

                )

            [2] => SimpleXMLElement Object
                (
                    [@attributes] => Array
                        (
                            [addressLine3] => teststring
                        )

                )

        )

)
`

**Comment by prewk** [comment]
Hi @stefanpausch!

My understanding was that both `<` and `>` were required to be escaped, but [it seems you are correct](https://stackoverflow.com/questions/5709232/how-do-i-include-etc-in-xml-attribute-values) about specifically `>` being exempt.

Could you perhaps make use of the `UniqueNode` parser instead? It's more resilient to "weird XML" in some regards (but only works if the node name is unique).

**Comment by prewk** [comment]
Yeah it's definitely a bug, so I will keep this issue open. I can't make any promises on when a fix can be made, though. Maybe you can preprocess your XML file as a quickfix with some clever `sed`/`awk` regexp or something for now.

**Comment by stefanpausch** [comment]
Take your time! - My problem is currently solved with UniqueNode and i will adapt my code to the different XML types i am parsing and will switch back to default XMLStringStreamer when your solution is ready. No need to rush things :)
